pydantic/types.py

```diff
@@ -516,3 +528,37 @@ def validate(cls, value: Union[str, bytes, int]) -> Union[IPv4Address, IPv6Addre
 
         with change_exception(errors.IPvAnyAddressError, ValueError):
             return IPv6Address(value)
+
+
+class IPvAnyInterface(_BaseAddress):
+    @classmethod
+    def __get_validators__(cls) -> 'CallableGenerator':
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, value: Union[str, bytes, int, NetworkType]) -> Union[IPv4Interface, IPv6Interface]:
```

Comment:
```suggestion
    def validate(cls, value: NetworkType) -> Union[IPv4Interface, IPv6Interface]:
```

Or am I missing something?